openedx/core/djangoapps/catalog/utils.py

```diff
@@ -19,7 +20,20 @@ def create_catalog_api_client(user, catalog_integration):
     return EdxRestApiClient(catalog_integration.internal_api_url, jwt=jwt)
 
 
-def get_programs(user, uuid=None, type=None):  # pylint: disable=redefined-builtin
+def _get_service_user(user, service_username):
+    """
+    Retrieve and return the Catalog Integration Service User Object
+    if the passed user is None or anonymous
+    """
+    if not user or user.is_anonymous():
+        try:
+            user = User.objects.get(username=service_username)
+        except User.DoesNotExist:
+            user = None
```

Comment:
Looks like we need to cover lines 29-32. https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/29174/Diff_Coverage_Report/